game/world/objects/npc.cpp

```diff
@@ -3091,9 +3113,19 @@ void Npc::setToFistMode() {
   }
 
 void Npc::aiPush(AiQueue::AiAction&& a) {
-  if(a.act==AI_OutputSvmOverlay)
-    aiQueueOverlay.pushBack(std::move(a)); else
-    aiQueue.pushBack(std::move(a));
+  switch(a.act) {
+    case AI_OutputSvmOverlay:
+      aiQueueOverlay.pushBack(std::move(a));
+      break;
+    case AI_QuickLook:
+      // QuickLook is supposed to last for 2 seconds
+      // and should not interrupt other AI actions like walking
+      quickLookEnd = owner.tickCount()+2000;
```

Comment:
State change should always occur when executing `Npc::nextAiAction`. Since `AI_QuickLook` should overlap with walking, `aiQueueOverlay` has to be utilized.
`